### PR TITLE
Document how to register an extension and how to extend the Graph

### PR DESCRIPTION
**Pull request by dpobel**
https://deploy-preview-776--heuristic-almeida-1a1f35.netlify.app/docs/remixed/guides/register-an-extension
https://deploy-preview-776--heuristic-almeida-1a1f35.netlify.app/docs/remixed/guides/extend-the-graphql-schema

### CODE REVIEW

File: docs/02-guides/extend-the-graphql-schema.mdx [hunk 1]
```diff
@@ -8,9 +8,284 @@ sidebar_position: 3
...
+
+The Front-Commerce core and platform integrations (such as Magento2) are
+implemented as GraphQL modules. They leverage features from the GraphQL Schema
+Definition Language (<abbr title="Schema Definition Language">SDL</abbr>).
+
+This page will guide you through the process of exposing a new feature in your
+GraphQL schema. We will create an extension with a  GraphQL module that allows
+to maintain a counter of clicks for a product.
+
+## Create an extension
```
Comment (Kagaminara) [inline docs/02-guides/extend-the-graphql-schema.mdx:26]: **thought:** Should this section be just a link to the `Register an extension` guide, or should both of them be merged at some point?

File: docs/02-guides/extend-the-graphql-schema.mdx [hunk 2]
```diff
@@ -3,14 +3,272 @@ title: Extend the GraphQL schema
...
+      const clickCountPromise = fetchClickCountForSku(product.sku);
+      return clickCountPromise
+    },
+  },
+
+  Mutation: {
+    incrementProductCounter: async (_root, params) => {
+      const { sku, incrementValue } = params;
+      try {
+        await incrementProducCount(sku, incrementValue);
```
Comment (real34) [inline docs/02-guides/extend-the-graphql-schema.mdx:261]:
```suggestion
        await incrementProductCount(sku, incrementValue);
```

File: docs/02-guides/extend-the-graphql-schema.mdx [hunk 3]
```diff
@@ -3,14 +3,272 @@ title: Extend the GraphQL schema
...
+
+As stated above, a resolver function can also return a Promise and thus handle
+asynchronous result. A more real life resolver map would like look like:
+
+```typescript title="extensions/click-counter/graphql/resolvers.ts"
+export default {
+  Product: {
+    clickCounter: (product) => {
+      const clickCountPromise = fetchClickCountForSku(product.sku);
+      return clickCountPromise
```
Comment (real34) [inline docs/02-guides/extend-the-graphql-schema.mdx:253]:
```suggestion
      return clickCountPromise;
```

**Review by real34 — APPROVED** [review]
:+1: It's clearer this way! Thanks.

**Pull request merged**